packages/apps/plugins/plugin-markdown/src/components/EditorMain.tsx

```diff
@@ -72,10 +72,21 @@ export const EditorMain = ({ id, readonly, toolbar, comments, extensions: _exten
     }
   });
 
+  // Toolbar actions.
+  const handleAction = useActionHandler(editorView);
   const [formattingState, formattingObserver] = useFormattingState();
+
+  const handleDrop: DNDOptions['onDrop'] = async (view, { files }) => {
+    const info = await fileManagerPlugin?.provides.file.upload?.(files[0]);
+    if (info) {
+      processAction(view, { type: 'image', data: info.url });
+    }
+  };
+
   const extensions = useMemo(() => {
     return [
       _extensions,
+      fileManagerPlugin?.provides.file.upload && dnd({ onDrop: handleDrop }),
```

Comment:
```suggestion
      fileManagerPlugin && dnd({ onDrop: handleDrop }),
```
this isn't being used as a type guard so the rest I don't think is necessary